File: backend/src/main/kotlin/com/plugga/backend/service/UserDeckService.java
package com.plugga.backend.service;

import com.plugga.backend.entity.UserDeck;

import java.util.List;

public interface UserDeckService {
    List<UserDeck> findAll();

    UserDeck findById(int id);

    List<UserDeck> findByDeckId(int id);

    List<UserDeck> findByUserId(int id);

    void save(UserDeck userCard);

    void deleteById(int id);
}

File: backend/src/main/kotlin/com/plugga/backend/service/UserDeckService.kt
package com.plugga.backend.service

import com.plugga.backend.entity.UserDeck
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface UserDeckService {
    fun findAll(pageable: Pageable): Page<UserDeck>

    fun findById(id: Int): UserDeck?

    fun findByDeckId(pageable: Pageable, id: Int): Page<UserDeck>

    fun findByUserId(pageable: Pageable, id: Int): Page<UserDeck>

    fun save(userDeck: UserDeck)

    fun deleteById(id: Int)
}
